server/src/main/java/org/opensearch/common/io/VersionedCodecStreamWrapper.java

```diff
@@ -46,10 +52,22 @@ public VersionedCodecStreamWrapper(IndexIOStreamHandler<T> indexIOStreamHandler,
      * @return stream content parsed into {@link T}
      */
     public T readStream(IndexInput indexInput) throws IOException {
-        CodecUtil.checksumEntireFile(indexInput);
-        int readStreamVersion = checkHeader(indexInput);
-        T content = getHandlerForVersion(readStreamVersion).readContent(indexInput);
-        return content;
+        logger.debug("Reading input stream [{}] of length - [{}]", indexInput.toString(), indexInput.length());
+        try {
+            CodecUtil.checksumEntireFile(indexInput);
+            int readStreamVersion = checkHeader(indexInput);
+            return getHandlerForVersion(readStreamVersion).readContent(indexInput);
+        } catch (CorruptIndexException cie) {
+            logger.error(
```

Comment:
If you log the exception and throw it as well, the stack trace will be logged twice. We can omit passing exception to the `logger.error`.